lib/elements/satin_column.py

```diff
@@ -747,8 +834,32 @@ def add_pair(pos0, pos1):
                     old_center = new_center
 
                 if to_travel <= 0:
-                    add_pair(pos0, pos1)
-                    to_travel = spacing
+                    ## add some randomness to the positions, but use unrandomned position to compute next ones
+                    randomizepos0=pos0
+                    randomizepos1=pos1
+                    decrease=0
+                    increase=0
+                    if self.random_first_rail_factor_in:
+                        decrease=self.random_first_rail_factor_in
+                    if self.random_first_rail_factor_out:
+                        increase=self.random_first_rail_factor_out
+                    if self.random_first_rail_factor_in or self.random_first_rail_factor_out:
+                        decalage0= random.uniform(-decrease/100,increase/100)
```

Comment:
I don't like the fact that this will produce different results between runs. Perhaps we should use a seeded (perhaps with a hash of the element ID) PRNG instead.

Reply:
I had been wondering that.... on the other hand it might be nice to be able to try again if  one is not happy with the first result.

Reply:
I agree with both of you. But once you are happy with the outcome, you will want you could keep it.
Maybe we can save the hash as an attribute to the element and add a button to recalculate?

Reply:
Now I'm thinking that we seed the PRNG from the element ID, but have a param to override the seed. That way we could try different seeds if we need to and then keep a result we're happy with.

As for PRNGs, I am going to try ChaCha20 as it is random-access with a 128-bit position field, enough to simply concatenate loop indexes together so that slight changes in the shape's geometry won't completely reorder random calls.